src/pipeline/cli_pipeline.c

```diff
@@ -0,0 +1,703 @@
+/**
+ * (C) Copyright 2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+#define D_LOGFAC DD_FAC(pipeline)
+
+#include <daos/common.h>
+#include <daos_pipeline.h>
+#include <daos/task.h>
+#include <daos_task.h>
+#include <daos/pool.h>
+#include <daos/container.h>
+#include <daos/object.h>
+#include <daos_types.h>
+#include <daos/dtx.h>
+#include <daos/placement.h>
+#include <daos/event.h>
+#include <daos/mgmt.h>
+#include <math.h>
+#include "pipeline_internal.h"
+#include "pipeline_rpc.h"
+
+struct pipeline_auxi_args {
+	int           opc;         /** Only one OPC for now (pipeline_run) */
+	uint32_t      map_ver_req; /** For I/O retry */
+	daos_obj_id_t omd_id;      /** I AM SETTING THIS BUT NOT USING IT */
+	tse_task_t   *api_task;    /** For I/O retry */
+	d_list_t      shard_task_head;
+};
+
+struct shard_pipeline_run_args {
+	uint32_t                   pra_map_ver; /** I AM SETTING THIS BUT NOT USING IT */
+	uint32_t                   pra_shard;
+	uint32_t                   pra_target;
+
+	daos_pipeline_run_t       *pra_api_args;
+	daos_unit_oid_t            pra_oid;
+	uuid_t                     pra_coh_uuid;
+	uuid_t                     pra_cont_uuid;
+	/*uint64_t			pra_dkey_hash; ??*/
+
+	struct pipeline_auxi_args *pipeline_auxi;
+};
+
+/** shard completion call back arguments */
+struct pipeline_run_cb_args {
+	uint32_t             shard;
+	crt_rpc_t           *rpc;
+	unsigned int        *map_ver; /** I AM SETTING THIS BUT NOT USING IT */
```

Comment:
please add like a TODO comment to do a pool map refresh to update this

Reply:
done